### PR TITLE
Homework 5 completed

### PR DESCRIPTION
**Pull request by Sapphireexe**
Finally completed that task with triangle and rhombus.

Every task in hw_5_1 finished.

Every task in hw_5_2 finished.

Every task in hw_5_3 finished.

### CODE REVIEW

File: HW_5/hw_5_1.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  Для Оксаны и ее друзей: также реализовать циклом while
+  Рекоммендация: попробуйте метод .repeat()
+*/
+
+let outputString = ':)';
+
+for (let i = 0; i < 5; i++) {
+    console.log(outputString);
+    outputString += ':)';
+};
```
Comment (anatoly-karpovich) [inline HW_5/hw_5_1.js]:
```suggestion
let outputString = ':)';
let result = '';

for (let i = 0; i < 5; i++) {
    result += outputString;
    console.log(result);   
};
```
With this implementation you can privide any other symbol without changing task logics

File: HW_5/hw_5_1.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+console.log('\n');
+
+//Вариант 2
+
+for (let i = 1; i <= 5; i++) {
+    console.log(':)'.repeat(i));
```
Comment (anatoly-karpovich) [inline HW_5/hw_5_1.js]: Same, use variable for a smile

File: HW_5/hw_5_1.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+};
+
+//Вариант 3
+
+console.log('\n');
+
+let iteration = 0;
+
+while (iteration < 5) {
+    console.log(':)'.repeat(iteration+1));
```
Comment (anatoly-karpovich) [inline HW_5/hw_5_1.js]: Once again)

File: HW_5/hw_5_2.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+console.log(sizeConverter(1_048_576)); // MB
+console.log(sizeConverter(1_048_575)); // kB
+console.log(sizeConverter(1024)); // kB
+console.log(sizeConverter(1023)); // B
+
+function sizeConverter (a) {
+  let result;
+  let byte = 1024;
+
+  if (isNaN(a)) {
```
Comment (anatoly-karpovich) [inline HW_5/hw_5_2.js]: Here we use only numbers, so just check for typeof number

File: HW_5/hw_5_2.js [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    return 'Input is not a number.'
+  };
+
+  if (a < byte) { // bytes
```
Comment (anatoly-karpovich) [inline HW_5/hw_5_2.js]: Where did you l lose the while loop?)

File: HW_5/hw_5_2.js [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+*/
+
+console.log(camelCaseConverter('I am super engineer'));
+console.log(camelCaseConverter('   I am super engineer   '));
+
+function camelCaseConverter (a) {
+  let result = '';
+
+  if (!a || typeof a !== 'string') {
+    return 'Error. Invalid input data.';
```
Comment (anatoly-karpovich) [inline HW_5/hw_5_2.js]:
```suggestion
    throw new Error("Invalid input data.");
```

File: HW_5/hw_5_3.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+console.log(vowelsConsonantsCounter('Incomprehensibleйцукен'));
+
+function vowelsConsonantsCounter (str) {
+  const vowels = 'eyuioa';
+  const consonants = 'qwrtpsdfghjklzxcvbnm';
+  let vowelsNumber = 0;
+  let consonantsNumber = 0;
+
+  if (! str || /[^a-zA-Z]/g.test(str)) {
+    return `Invalid input string. Must contain word only.`
```
Comment (anatoly-karpovich) [inline HW_5/hw_5_3.js]:
```suggestion
    throw new Error(`Invalid input string. Must contain word only.`)
```

File: HW_5/hw_5_3.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+console.log(caesarCipher('AaBbXxZz', 1));
+console.log(caesarCipher('AaBbXxZz', -1));
+console.log(caesarCipher('AbC', -1));
+console.log(caesarCipher('ZzZ', 1));
+
+function caesarCipher (str, offset) {
+  let encryptedString = '';
+
+  if (! str || /[^a-zA-Z]/g.test(str)) {
+    return `Invalid input string. Must contain word only.`
```
Comment (anatoly-karpovich) [inline HW_5/hw_5_3.js]: Same logic for errors as above

**Review by anatoly-karpovich — CHANGES_REQUESTED** [review]

**Comment by Sapphireexe** [comment]
Made changes to the task according to the notes

**Review by anatoly-karpovich — DISMISSED** [review]

**Review by VladimirLQA — DISMISSED** [review]

**Comment by Sapphireexe** [comment]
Force-merged hw_5 into main because of two MR conflicts. Will be more attentive.

**Pull request merged**